### ISSUE TITLE
Illegal modification of inherited element in Modelica.Media.Water.WaterIF97_base

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#505]
**Reported by per&#x2e;ostlund on 15 Mar 2011 12:39 UTC**
The Modelica.Media.Water.WaterIF97_base package uses class extends to extend the BaseProperties model that it inherits from Modelica.Media.Interfaces.PartialTwoPhaseMedium. When doing so it declares an element sat that has modifications, while at the same time inheriting an element sat from BaseProperties in PartialTwoPhaseMedium that does not have any modifications. Since the two sat elements are not identical due to different modifications this shouldn't be legal according to 7.1 in the Modelica 3.2 specification. Here's a simplification of the code from the MSL 3.2 (it's the same in MSL 3.1 too):
```
partial package PartialTwoPhaseMedium
  extends PartialPuseSubstance;

  redeclare replaceable partial model extends BaseProperties
    SaturationProperties sat;
  end BaseProperties;
end PartialTwoPhaseMedium;

partial package WaterIF97_base
  extends PartialTwoPhaseMedium;

  redeclare replaceable model extends BaseProperties
    SaturationProperties sat(Tsat(start = 300.0), psat(start = 1.0e5));
  end BaseProperties;
end WaterIF97_base;
```
Dymola does not give any error for this (but it does if you modify the component directly instead of modifying it's subcomponents), but this causes problems for OpenModelica since we do give an error in this case. Have I overlooked something, or is this indeed actually illegal?

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/505

**Comment by modelica-trac-importer** [comment]
**Modified by sjoelund&#46;se on 31 Mar 2011 11:01 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by hubertus on 21 Apr 2011 12:49 UTC**
Fixed in d25ad060c1202b5713737cfad9b0e8faa278ab38 (trunk) and c252ea7e9180110729a94a0802d7c3c6f3f0b138 (3.1 maintenance)

**Comment by modelica-trac-importer** [comment]
**Modified by dietmarw on 21 Apr 2011 19:08 UTC**
